### PR TITLE
Intro to IoT Security

### PR DESCRIPTION
**Pull request by JohnPWSharp**
connects to #38

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@JohnPWSharp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by bennage** [comment]
@MikeWasson I'd be interested in having your feedback on this.

**Comment by tomkerkhove** [comment]
Is it possible that the articles are not publically available on the repo? I'd love to have a look at it!

**Comment by bennage** [comment]
@tomkerkhove The article is visible as part of this pull request. Look at the Files Changed tab above. After it's merged, it will be in the master branch. Before it's merged you can comment it on the files line-by-line.

File: docs/articles/introduction-to-IoT-security.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+In an IoT solution, security and safety are commonly intertwined. In a business environment, data emanating from business processes is commercially valuable and is often a target for large-scale industrial espionage. Software developers have traditionally been more concerned with hardening systems to make the action of mounting an attack commercially unattractive (it can cost a third party more in resources to break into the system and steal information than is profitable). Software engineers constructing IoT solutions have much broader security concerns. Many IoT systems control critical operations and the core of industrial and civil infrastructure. These are prime targets for politically motivated attacks (terrorism)rather than those driven by profit. Consequently, digital security will be increasingly interwoven with the physical safety of life and equipment. This should provide deep motivation for ensuring that security is addressed in a robust and thorough manner.
```
Comment (bennage) [inline docs/articles/introduction-to-IoT-security.md]: I like the opening theme of this paragraph: how security and safety are intertwined concerns. However, the rest of the paragraph quickly diverges.

I also think that our examples tend toward things that people already view as critical (public services, infrastructure, etc.) Part of the surprise for me in researching IoT and security is that seemingly innocuous systems can present safety concerns (e.g. sending a malicous document to a printer and starting a fire).

I'd like to hook the reader quickly with some kind of "here's some things that might surprise you".

File: docs/articles/introduction-to-IoT-security.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+In this diagram, devices are the individual items of hardware that are located in the field. They can have varying levels of complexity, but typically report status information and may respond to commands. A field gateway is a device, appliance, or more general-purpose server that acts as a communications enabler, device control system, or local data processing hub. As an example, a smart car could be considered to be a field gateway with its own local control software gathering data from individual devices located on the vehicle, monitoring them, and sending commands back to them. A field gateway, or individual devices, might need to communicate with a remote control system that controls and coordinates actions across different sites. This could be achieved by connecting to the services that implement this control system directly or through a cloud gateway. A cloud gateway can act as a regional hub implementing some autonomous regional services. The connection between a field gateway and/or local devices with a cloud gateway could be static or dynamic. For example, a field gateway that roams (such as a smart car) might switch to different cloud gateways as it changes location.
```
Comment (bennage) [inline docs/articles/introduction-to-IoT-security.md]: This seems like a lot of introductory material isn't directly related to security. I would prefer to tigten this up and link to supporting content.

File: docs/articles/introduction-to-IoT-security.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+Even if a device is physically secured, spoofing and impersonation could still occur, especially if communications with a field gateway are wireless. Many devices are cheap commodity items that do not include the capability to encrypt data so it could be easy to introduce a rogue device that assumes the identity of a valid one into the local network and starts to present bad data. Additionally, a DoS attack could be staged by using a transmitter that interferes with the frequency used by the local wireless network without requiring any other form of entry into the system. Furthermore, it could be possible for an observer to silently intercept traffic, resulting in disclosure of sensitive or valuable information. Some form of shielding may be necessary.
+
+Devices that listen for incoming connections can be especially vulnerable unless these connections are authenticated in some way. Even then, a device can be made the target of a DoS attack simply by bombarding it with a large number of connection requests; even if these requests are recognized as invalid by the device, it can still be overwhelmed by the processing involved.
```
Comment (bennage) [inline docs/articles/introduction-to-IoT-security.md]: This is very true. In fact, this is so universal that we _will_ give some specific guidance here. This is what @clemensv has named "Service Assisted Communication". If we introduce this risk here, let put in a placeholder that should link to the pattern. I've added an issue #168.

File: docs/articles/introduction-to-IoT-security.md (hunk 1 above)
Comment (JohnPWSharp) [inline docs/articles/introduction-to-IoT-security.md]: I have trimmed this paragraph down to focus on the important issues.

File: docs/articles/introduction-to-IoT-security.md (hunk 2 above)
Comment (JohnPWSharp) [inline docs/articles/introduction-to-IoT-security.md]: I have cut this back and referenced the "What is an IoT Solution?" doc.

File: docs/articles/introduction-to-IoT-security.md (hunk 3 above)
Comment (JohnPWSharp) [inline docs/articles/introduction-to-IoT-security.md]: I have added a placeholder

**Pull request merged**